core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -495,10 +496,16 @@ private RuntimeException imageGenerationFailed(int exitValue, List<String> comma
     private void checkGraalVMVersion(GraalVM.Version version) {
         log.info("Running Quarkus native-image plugin on " + version.getFullVersion());
         if (version.isObsolete()) {
-            final int major = GraalVM.Version.CURRENT.major;
-            final int minor = GraalVM.Version.CURRENT.minor;
-            throw new IllegalStateException("Out of date version of GraalVM detected: " + version.getFullVersion() + "."
-                    + " Quarkus currently supports " + major + "." + minor + ". Please upgrade GraalVM to this version.");
+            String currentVersion = GraalVM.Version.CURRENT.getFullVersion();
+            if (GraalVM.Version.CURRENT != GraalVM.Version.MINIMUM) {
+                currentVersion += " (minimum: " + GraalVM.Version.MINIMUM.getFullVersion() + ")";
+            }
+            String msg = String.format("Out of date version of GraalVM detected: %s."
+                    + " Quarkus currently supports %s. Please upgrade to %s.",
+                    version.getFullVersion(),
+                    currentVersion,
+                    GraalVM.Version.CURRENT.getFullVersion());
```

Comment:
I'm not really happy with the wording change. I would keep the message as is, just using CURRENT and major.minor.

I prefer recommending only one version even if we don't force the upgrade.

Reply:
Just to be clear: you recommend to display only the current version in the message and use the minimum just to check if the GraalVM used is obsolete?

Reply:
Yes, I would keep the current message as is. And use the minimum version to determine if we should display it.

That's what we did before Galder's refactoring. This way we encourage people to migrate to the currently supported (and tested) version.